build.js

```diff
@@ -1,34 +1,125 @@
+const assert = require("assert");
+const child_process = require("child_process");
+const { version } = require("./package.json");
 
-const vscodeVersion = "1.84.2";
+const VSCODE_VERSION = version.split("-")[0];
+
+function error(msg) {
+  console.info("\x1b[31merror %s\x1b[0m", msg)
+}
+function ok(msg) {
+  console.info("\x1b[32m%s\x1b[0m", msg)
+}
+function note(msg) {
+  console.info("\x1b[90m%s\x1b[0m", msg)
+}
+function exec(cmd, opts) {
+  console.info("\x1b[36m%s\x1b[0m", cmd)
+  return child_process.execSync(cmd, opts);
+}
+
+const requiredTools = ["node", "yarn", "git", "python"];
+note(`required tools ${JSON.stringify(requiredTools)}`)
+for (const tool of requiredTools) {
+  try {
+    child_process.execSync(`${tool} --version`, { stdio: "ignore" });
+  } catch (e) {
+    error(`"${tool}" is not available.`);
+    process.exit(1);
+  }
+}
+ok("required tools installed")
+
+const node_version_out = child_process.execSync(`node -v`);
+const node_version = node_version_out.toString().trim()
+if (node_version < "v20.0") {
+  error(`Want node > 20. Got "${node_version}"`);
+  process.exit(1);
+}
 
 if (!fs.existsSync("vscode")) {
-  child_process.execSync(`git clone --depth 1 https://github.com/microsoft/vscode.git -b ${vscodeVersion}`, {
+  note("cloning vscode")
+  exec(`git clone --depth 1 https://github.com/microsoft/vscode.git -b ${VSCODE_VERSION}`, {
     stdio: "inherit",
   });
+} else {
+  ok("vscodium already installed")
+  note("delete vscodium folder to clone again")
 }
+
+note("changing directory to vscode")
 process.chdir("vscode");
 
 if (!fs.existsSync("node_modules")) {
-  child_process.execSync("yarn", { stdio: "inherit" });
+  exec("yarn", { stdio: "inherit" });
+} else {
+  ok("node_modules exists. Skipping yarn")
 }
+
 // Use simple workbench
-fs.copyFileSync(
-  "../workbench.ts",
-  "src/vs/code/browser/workbench/workbench.ts"
-);
+note("copying workbench file")
+fs.copyFileSync("../workbench.ts", "src/vs/code/browser/workbench/workbench.ts");
 
 // Compile
-child_process.execSync("yarn gulp vscode-web-min", { stdio: "inherit" });
+note("starting compile")
+exec("yarn gulp vscode-web-min", { stdio: "inherit" });
+ok("compile completed")
 
 // Extract compiled files
 if (fs.existsSync("../dist")) {
-  fs.rmdirSync("../dist", { recursive: true });
+  note("cleaning ../dist")
+  fs.rmSync("../dist", { recursive: true, force: true });
+} else {
+  ok("../dist did not exist. No need to clean")
 }
+
 fs.mkdirSync("../dist");
-fse.copySync("../vscode-web", "../dist");
+fs.cpSync("../vscode-web", "../dist", { recursive: true, force: true });
+ok("copied ../vscode-web to ../dist")
+
+// Patch output
+note("beginning complex patches for networking")
+const distWorkbenchPath = "../dist/out/vs/workbench/workbench.web.main.js";
+const workbench = fs.readFileSync(distWorkbenchPath, "utf8");
+const workbenchPatched = workbench.replace(/("https:\/\/{{uuid}}[^"]+")/g, "(globalThis.VSCODE_WEB_PUBLIC_URL||$1)");
```

Comment:
I don't understand the point of this variable VSCODE_WEB_PUBLIC_URL
what url do you need to replace ?

Reply:
I just took that from the fork of zorse (https://github.com/zorse-code/vscode-web) and thought: more flexibility probably doesn't hurt 😅 

In the fork it is mentioned that:

This fork takes in a configuration parameter window.VSCODE_WEB_PUBLIC_URL. If this is set it is used instead of vscode-cdn.net for all iframe and web worker requests.

But in my tests i didn't use it.
This is for sure not necessary.

Reply:
Ok, I found the purpose, it seems webview are using a cdn url to provide some content.
This url is a parameter in `product.json`

```json
{
  "webviewContentExternalBaseUrlTemplate": "https://{{uuid}}.vscode-cdn.net/insider/ef65ac1ba57f57f2a3961bfe94aa20481caca4c6/out/vs/workbench/contrib/webview/browser/pre/"
}
```

I think it should not be necessary to override it on compiled file, using the property in product.json should be enough